### PR TITLE
Vendors serving v0.9.0 

### PR DESCRIPTION
**Pull request by navidshaikh**
/lint

## Proposed Changes

- Vendors serving v0.9.0
- Updates code and tests to refer the required data structure and method/function calls
- duckv1beta1 -> duckv1 conditions
- traffic percent value is now *int64
- containerConcurrency validation method relocation

### CODE REVIEW

**Comment by navidshaikh** [comment]
@rhuss : Yes, looking at it, apparently I introduced a bug in `RemoveNullTargets` method and we are seeing the difference in actual vs expected targets.

File: pkg/serving/config_changes.go [hunk 1]
```diff
@@ -24,9 +23,12 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"knative.dev/pkg/ptr"
 	"knative.dev/serving/pkg/apis/autoscaling"
 	servingv1alpha1 "knative.dev/serving/pkg/apis/serving/v1alpha1"
-	servingv1beta1 "knative.dev/serving/pkg/apis/serving/v1beta1"
+
+	//servingv1beta1 "knative.dev/serving/pkg/apis/serving/v1beta1"
```
Comment (toVersus) [inline pkg/serving/config_changes.go]: Can we remove this line?

File: pkg/serving/config_changes_test.go [hunk 2]
```diff
@@ -62,8 +63,8 @@ func TestUpdateInvalidAutoscalingAnnotations(t *testing.T) {
 	if annos[autoscaling.TargetAnnotationKey] != "1000" {
 		t.Error("target failed")
 	}
-	if template.Spec.ContainerConcurrency != 1000 {
-		t.Error("limit failed")
+	if *template.Spec.ContainerConcurrency != 1000 {
+		t.Errorf("limit failed")
```
Comment (toVersus) [inline pkg/serving/config_changes_test.go]:
```suggestion
		t.Error("limit failed")
```

File: pkg/serving/config_changes_test.go [hunk 3]
```diff
@@ -492,7 +493,7 @@ func getV1alpha1RevisionTemplateWithOldFields() (*servingv1alpha1.RevisionTempla
 	container := &corev1.Container{}
 	template := &servingv1alpha1.RevisionTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "template-foo",
+			Name:      "-template-foo",
```
Comment (toVersus) [inline pkg/serving/config_changes_test.go]: I'm not sure why we need to put hyphen as a prefix.

**Pull request merged**